app/src/main/java/com/example/marina/openweather/screens/main/MainActivity.java

```diff
@@ -34,69 +34,78 @@
     RecyclerView recyclerView;
     @BindView(R.id.fab)
     FloatingActionButton fab;
+    @BindView(R.id.progressBar)
+    ProgressBar progressBar;
 
-    @Inject
-    Retrofit retrofit;
-    @Inject
-    OkHttpClient okHttpClient;
-    @Inject
-    Api api;
-
-    CityAdapter adapter;
-    List<City> cities = new ArrayList<>();
-
-
+    private CityAdapter adapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
         setSupportActionBar(toolbar);
 
-        MyApplication.get().getComponent().inject(this);
-        bind();
-
-        fab.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-               // mainPresenter.getWeather();
-                MainDialog mainDialog = new MainDialog(MainActivity.this);
-                mainDialog.show();
-
-            }
+        fab.setOnClickListener(view -> {
+            mainPresenter.showAlert();
         });
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
-
     }
+
     @Override
-    public void onResume(){
+    public void onResume() {
         super.onResume();
-        mainPresenter.getWeather();
+        mainPresenter.getWeather("Kharkiv");
     }
 
     @Override
-    public void showCities(List<City> cities) {
-        this.cities = cities;
-        if(adapter == null) {
-            adapter = new CityAdapter(this.cities);
+    public void showMessage(int idMessage) {
+        Snackbar.make(fab, getString(idMessage), Snackbar.LENGTH_LONG)
+                .setAction("Action", null).show();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        mainPresenter.unSubscribe();
+    }
+
+    @Override
+    public void setData(List<Response> cities) {
+        if (adapter == null) {
```

Comment:
For consideration in further development: it would be better to initialize adapter at start, and then just update the dataset